## Question One (Mean):

The first question is to find the mean of a set of numbers. From class, we understood the mean is calculated by 
$$\begin{equation}
\mu = \dfrac{1}{n}\sum_{i=1}^{n}x_{i}
\end{equation}$$

So using the formulae, write a function that would calcuate the mean

In [4]:
def meanOfNumbers(numbers):
    sum = 0
    for i in numbers:
        sum += i
    result = sum / len(numbers)
    return(result)

num1= [19, 21, 46, 11, 18]
num2 = [45, 67, 89, 12, 31, 25]

meanOfNumbers(num2)



44.833333333333336

Use your function to answer the following questions:


Q1. [19, 21, 46, 11, 18]= 23.0

Q2. [45, 67, 89, 12, 31, 25] = 44.833333333333336



## Question Two (Median):

You would have to find the median of a given list. 
Remember: The median is the middle number for a given list. But, note that incase the length of the list is even you calculate it by taking the average of the two middle numbers.

In [8]:
def middle(num):
    num.sort()
    x = len(num)
    if x%2 == 0:
        mid = x//2
        midFinal = num[mid-1] + num[mid]
        result = midFinal/2
        return(result)
    else:
        mid = (x+1)//2
        midFinal = mid-1
        result = num[midFinal]
        return(result)

numbers = [21, 3, 7, 17, 19, 31, 46, 20 ,43 ]
middle(numbers)

20


Use your function to find the median of the following list:

Q1. 15, 14, 10, 8, 12, 8 and 16 = 12

Q2. 21, 3, 7, 17, 19, 31, 46, 20 and 43. = 20

## Question Three (Mode):

Mode is the number in the list which occurs most frequently. We can have two or three modes in a list if they equal the same number of frequency. In that, case, your function should return all of the most frequently occured numbers in a list.

Find the  mode of the following using your function:

    
Q1. 2, 3, 6,6, 6, 3 

Q2. "a", "a", "b" and "d"



In [14]:
from collections import Counter

def mode9(num):
    numCount = Counter(num)
    req = max(numCount.values())
    result = []
    for val,count in numCount.items():
        if count==req:
            result.append(val)
    return(result)

mode9(["a", "a", "b" , "b", "d"])

['a', 'b']

## Question Four (Variance):

For this part, you need to make use of the original written calculated mean function to calculate the variance.
$$ \begin{equation}
\sigma ^{2} = \dfrac{1}{n}\sum_{i=1}^{n} (x_{i}- \mu)^{2}
\end{equation}$$


Use your variane function to find the following:

Q1.  6, 7, 3, 9, 10 and 15 

Q2. 10, 32, 45, 67, 31 and 73

## Question Five (Standard deviation):

For this part, you need to make use of the original written calculated mean function to calculate the standard deviation.



$$ \begin{equation}
\sigma = \sqrt{\dfrac{1}{n}\sum_{i=1}^{n} (x_{i}- \mu)^{2}}
\end{equation}$$

Find the standard deviation of the following:
    
Q1. 9, 2, 5, 4, 12 and 7


Q2. 9, 2, 5, 4, 12, 7, 8, 11, 9, 3, 7, 4, 12, 5, 4, 10, 9, 6, 9, 4

## Question Six (Range):
Range is the difference between the lowest and highest value.

Find the range of the following list:

Q1. 9, 2, 5, 4, 12, and  7 


Q2. 9, 2, 5, 4, 12, 7,1, 8, 11, 9, 3, 7, 4

    